src/utils/functionRuntimeUtils.ts

```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as opn from 'opn';
+import * as vscode from 'vscode';
+import { extensionPrefix } from '../../src/ProjectSettings';
+import { DialogResponses } from '../DialogResponses';
+import { localize } from '../localize';
+import { cpUtils } from './cpUtils';
+import { nodeUtils } from './nodeUtils';
+
+export namespace functionRuntimeUtils {
+
+    const runtimePackage: string = 'azure-functions-core-tools';
+
+    export async function validateFunctionRuntime(outputChannel: vscode.OutputChannel): Promise<void> {
+        const errorMessage: string | undefined = await getRuntimeValidationError(outputChannel);
+        if (errorMessage) {
+            await promptDocumentationAction(errorMessage);
+        }
+    }
+
+    async function getRuntimeValidationError(outputChannel: vscode.OutputChannel): Promise<string | undefined> {
+        try {
+            await cpUtils.executeCommand(outputChannel, undefined, 'func');
+        } catch (error) {
+            return localize(
+                'azFunc.noFunctionRuntime',
+                '[Azure Functions] Azure-functions-core-tools is not installed. Please install it otherwise Azure Functions extension may fail.'
```

Comment:
I don't think we should display this warning or the npm warning below for two reasons:
1. Most importantly, users don't need the func cli to perform many operations (create new project, create new function, deploy, view streaming logs, etc.) so the fewer the warnings about this, the better
1. Less importantly, this extension is included in a few extension packs like [this](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-node-azure-pack) and we don't want negatively impact the experience of the pack as a whole (aka maybe they want to use the azure app service stuff, but not the functions stuff)

I think we should _only_ display a warning if the runtime is out-of-date, otherwise swallow all errors.

Reply:
Make sense